### PR TITLE
Recomend the Java related runtime in list

### PR DESCRIPTION
**Pull request by jdneo**
Signed-off-by: Sheng Chen <sheche@microsoft.com>

### CODE REVIEW

File: appservice/src/createAppService/createWebApp.ts [hunk 1]
```diff
@@ -23,28 +23,38 @@ export async function createWebApp(
 }
 
 export async function setWizardContextDefaults(wizardContext: IAppServiceWizardContext, actionContext: IActionContext, advancedCreation?: boolean): Promise<void> {
+    const isJavaProject: boolean = await javaUtils.isJavaProject();
+    if (isJavaProject) {
```
Comment (jdneo) [inline appservice/src/createAppService/createWebApp.ts]: @EricJizbaMSFT Do you think we need to enable the multi-root for `js` & `python` here?

The current implementation is separated.

File: appservice/src/createAppService/createWebApp.ts [hunk 2]
```diff
@@ -23,28 +23,38 @@ export async function createWebApp(
...
+        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime ? [...wizardContext.recommendedSiteRuntime.keys()].join(',') : undefined;
     }
 
     if (!advancedCreation) {
         await LocationListStep.setLocation(wizardContext, 'centralus');
+        if (isJavaProject) {
+            // considering high resource requirement for Java applications, a higher plan sku is set here
+            wizardContext.newPlanSku = { name: 'P1v2', tier: 'PremiumV2', size: 'P1v2', family: 'P', capacity: 1 };
+            // to avoid 'Requested features are not supported in region' error
+            await LocationListStep.setLocation(wizardContext, 'westeurope');
```
Comment (jdneo) [inline appservice/src/createAppService/createWebApp.ts]: I found the `P1v2` is not always available in `Central US`. I've met this problem, but not stable repro.

So the location is set to `West Europe` here.

**Comment by jdneo** [comment]
Hi @EricJizbaMSFT,

Would you mind to take a look at this PR?

For the issue that P1v2 may fail to provision, I've sent emails to the service team and waiting for the response.

File: appservice/src/createAppService/createWebApp.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply appservice/src/createAppService/createWebApp.ts]: I'm not worried about multi-root. According to VS Code folks it has pretty low usage

File: appservice/src/createAppService/SiteRuntimeStep.ts [hunk 3]
```diff
@@ -167,15 +167,19 @@ export class SiteRuntimeStep extends AzureWizardPromptStep<IAppServiceWizardCont
         ];
     }
 
-    private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ILinuxRuntimeStack>[], runtime: string): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
+    private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ILinuxRuntimeStack>[], recomendedRuntimes: Map<LinuxRuntimes, number>): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
```
Comment (EricJizbaMSFT) [inline appservice/src/createAppService/SiteRuntimeStep.ts]: The map seems over-complex. What if it was just a sorted array of recommendedRuntimes. For example, this function could be implemented like this:
```typescript
private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ILinuxRuntimeStack>[], recommendedRuntimes: string[]): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
    function getPriority(item: IAzureQuickPickItem<ILinuxRuntimeStack>): number {
        const index: number = recommendedRuntimes.findIndex(runtime => item.data.name.includes(runtime));
        return index === -1 ? recommendedRuntimes.length : index;
    }
    return runtimeItems.sort((a, b) => getPriority(a) - getPriority(b));
}
```

File: appservice/src/createAppService/createWebApp.ts (hunk 2 above)
Comment (EricJizbaMSFT) [reply appservice/src/createAppService/createWebApp.ts]: Sounds fine to me. I don't have a strong preference on default location.

File: appservice/src/createAppService/SiteRuntimeStep.ts (hunk 3 above)
Comment (jdneo) [reply appservice/src/createAppService/SiteRuntimeStep.ts]: Cool, using `array` is much better!

File: appservice/src/createAppService/createWebApp.ts (hunk 2 above)
Comment (jdneo) [reply appservice/src/createAppService/createWebApp.ts]: Ok. I'll revert the code for switching location once the issue has been addressed.

File: appservice/src/createAppService/createWebApp.ts [hunk 4]
```diff
@@ -23,28 +23,38 @@ export async function createWebApp(
...
         if (await fse.pathExists(path.join(fsPath, 'package.json'))) {
-            wizardContext.recommendedSiteRuntime = LinuxRuntimes.node;
+            wizardContext.recommendedSiteRuntime = [ LinuxRuntimes.node ];
         } else if (await fse.pathExists(path.join(fsPath, 'requirements.txt'))) {
             // requirements.txt are used to pip install so a good way to determine it's a Python app
-            wizardContext.recommendedSiteRuntime = LinuxRuntimes.python;
+            wizardContext.recommendedSiteRuntime = [ LinuxRuntimes.python ];
         }
-        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime;
+        actionContext.properties.recommendedSiteRuntime = wizardContext.recommendedSiteRuntime ? [...wizardContext.recommendedSiteRuntime.keys()].join(',') : undefined;
```
Comment (EricJizbaMSFT) [inline appservice/src/createAppService/createWebApp.ts]: Simplify this line to work for an array instead of a map, but otherwise looks good

Comment (jdneo) [reply appservice/src/createAppService/createWebApp.ts]: Oh sorry, I missed this part. Updated...

**Pull request merged**